### PR TITLE
Handle NSItemProvider public.image data types.

### PR DESCRIPTION
**Pull request by stefanceriu**

### CODE REVIEW

File: ElementX/Sources/Other/Extensions/NSItemProvider.swift [hunk 1]
```diff
@@ -6,23 +6,134 @@
...
+        
+        switch contentType.identifier {
+        case UTType.image.identifier:
+            return "png"
+        default:
+            return nil
+        }
+    }
+    
+    private func isIdentifierSupported(_ identifier: String?) -> Bool {
```
Comment (pixlwave) [inline ElementX/Sources/Other/Extensions/NSItemProvider.swift:135]: Would be nice to add a comment here saying something like

> Don't filter out generic `public.image` content as screenshots are in this format and we can convert it to a PNG ourselves.

File: ElementX/Sources/Other/Extensions/NSItemProvider.swift [hunk 2]
```diff
@@ -6,23 +6,134 @@
...
+            }
+        }
+        
+        guard let shareData else {
+            MXLog.error("Invalid share data")
+            return nil
+        }
+        
+        do {
+            if let suggestedName = (suggestedName as? NSString),
```
Comment (pixlwave) [inline ElementX/Sources/Other/Extensions/NSItemProvider.swift:83]:
```suggestion
            if let suggestedName = suggestedName as? NSString,
```

File: ElementX/Sources/Other/Extensions/NSItemProvider.swift [hunk 3]
```diff
@@ -6,23 +6,134 @@
...
+            return nil
+        }
+        
+        guard let pngData = uiImage.pngData() else {
+            MXLog.error("Failed extracting PNG data out of the UIImage")
+            return nil
+        }
+        
+        do {
+            if let suggestedName = (suggestedName as? NSString),
```
Comment (pixlwave) [inline ElementX/Sources/Other/Extensions/NSItemProvider.swift:43]:
```suggestion
            if let suggestedName = suggestedName as? NSString,
```

File: ElementX/Sources/Other/Extensions/NSItemProvider.swift [hunk 4]
```diff
@@ -6,23 +6,134 @@
...
+        
+        guard let shareData else {
+            MXLog.error("Invalid share data")
+            return nil
+        }
+        
+        do {
+            if let suggestedName = (suggestedName as? NSString),
+               // Suggestions are nice but their extension is `jpeg`
+               let filename = (suggestedName.deletingPathExtension as NSString).appendingPathExtension(contentType.fileExtension) {
```
Comment (pixlwave) [inline ElementX/Sources/Other/Extensions/NSItemProvider.swift:85]: Double checking here as I thought the old logic was working fine outside of screenshots: Isn't this useless jpeg suggestion only for public.image?

**Review by pixlwave — APPROVED** [review]
Nice 👌

File: ElementX/Sources/Other/Extensions/NSItemProvider.swift (hunk 4 above)
Comment (stefanceriu) [reply ElementX/Sources/Other/Extensions/NSItemProvider.swift:85]: Sure is, sorry, copy pasta 🤦‍♂️

**Pull request merged**